gcovr/html_generator.py

```diff
@@ -115,6 +119,41 @@ def render(options):
         )
 
 
+class CodeFormater():
+
+    formatter = None
+    lexer = None
```

Comment:
I think this class is too complicated:

1. It models two entirely separate cases: one where the `formatter` is set, and one where the formatter is None. This looks like a good candidate for the [Replace Conditionals with Polymorphism](https://refactoring.com/catalog/replaceConditionalWithPolymorphism.html) refactoring, or for a Null Object or something.
2. Introducing a stateful `lexer` variable is potentially less clear than directly returning a function that highlights the code.

As a rough draft I'd do something like this:

```python
class NullHighlighting:
  def css(self):
    return ''

  @staticmethod
  def highlighter_for_file(filename):
    return lambda code: code

class PygmentHighlighting:
  def __init__(self):
    self.formatter = HtmlFormatter()

  def css(self):
    return self.formatter.get_style_defs()

  def highlighter_for_file(self, filename):
    try:
      lexer = get_lexer_for_filename(fielname)
    except pygments.util.ClassNotFound:
      return NullHighlighting.highlighter_for_file(filename)
    return lambda code: pygments.highlight(code, lexer, self.formatter)
```

Used as:

```python
highlighting = PygmentHighlighting() if options.html_details_with_highlighting else NullHighlighting()
hl = highlighting.highlighter_for_file(filename)
code = hl(code).rstrip()
```

Reply:
I'll change this together with the lazy loading and wrap the string into the Markup